js/bitrue.js

```diff
@@ -1676,6 +1677,71 @@ module.exports = class bitrue extends Exchange {
         return this.parseTransactions (data, currency);
     }
 
+    async fetchTransactionFees (codes = undefined, params = {}) {
+        /**
+         * @method
+         * @name bitrue#fetchTransactionFees
+         * @description fetch transaction fees
+         * @see https://github.com/Bitrue-exchange/Spot-official-api-docs#exchangeInfo_endpoint
+         * @param {[string]|undefined} codes list of unified currency codes
+         * @param {object} params extra parameters specific to the bitrue api endpoint
+         * @returns {object} a list of [fee structures]{@link https://docs.ccxt.com/en/latest/manual.html#fee-structure}
+         */
+        await this.loadMarkets ();
+        const response = await this.v1PublicGetExchangeInfo (params);
+        const coins = this.safeValue (response, 'coins');
+        return this.parseTransactionFees (coins, codes);
+    }
+
+    parseTransactionFees (response, codes = undefined) {
+        const result = {};
+        for (let i = 0; i < response.length; i++) {
+            const entry = response[i];
+            const currencyId = this.safeString (entry, 'coin');
+            const currency = this.safeCurrency (currencyId);
+            const code = this.safeString (currency, 'code');
+            if ((codes === undefined) || (this.inArray (code, codes))) {
+                result[code] = this.parseTransactionFee (entry);
+            }
+        }
+        return result;
+    }
+
+    parseTransactionFee (transaction) {
+        //
+        // {
+        //     "coin": "egc",
+        //     "coinFulName": "EverGrow",
+        //     "chains": [
+        //         "BSC"
+        //     ],
+        //     "chainDetail": [
+        //         {
+        //             "chain": "BSC",
+        //             "enableWithdraw": true,
+        //             "enableDeposit": false,
+        //             "withdrawFee": "2500000.0000000000",
+        //             "minWithdraw": "5000000.0000000000",
+        //             "maxWithdraw": "10000000000000000.0000000000"
+        //         }
+        //     ]
+        // }
+        //
+        const result = {
+            'withdraw': {},
+            'deposit': {},
+            'info': transaction,
+        };
```

Comment:
It looks like every other exchange has these keys one level up in the dictionary, so they would be returned in the response of `fetchTransactionFees`, while I like these fields being returned individually for each currency, we should have all the responses match each other and I don't know how @carlosmiei @kroitor and @frosty00 feel about changing every other exchange at this point

Reply:
@samgermain so basically u're suggesting indexing it by currency?

```Javascript
{
    "USDT": {
           "deposit":...
           "withdraw":...
            "info"...
    }
}

Reply:
like @AlgoQ has done in a couple of exchanges right?
eg: https://github.com/ccxt/ccxt/pull/15439

Reply:
> @samgermain so basically u're suggesting indexing it by currency?
> 
> ```js
> {
>     "USDT": {
>            "deposit":...
>            "withdraw":...
>             "info"...
>     }
> }
> ```

It looks like that's how it's done here, but when I looked at other exchanges, they return one `info` key with all the information in it one level higher, I didn't check every exchange. I'm only suggesting this way for consistency, but I think having info per currency is better

Reply:
@samgermain I agree and the response is much cleaner when the user provides only one currency

Reply:
> @samgermain I agree and the response is much cleaner when the user provides only one currency

So we're going to use this format then?

Reply:
It turns out our response is not very unified for this method, I'll have to go through and fix this

Reply:
@samgermain I intended to open a discussion to change all implementations to match this new structure, wdyt?

Reply:
> @samgermain I intended to open a discussion to change all implementations to match this new structure, wdyt?

i think we should just use the structure with `{'code': {'withdrawals': [], 'deposits': []}}`, the only downside is binance uses `{'withdrawals': {'code': []}, 'deposits': {'code': []}}`

Reply:
@kroitor can you give your take on this? We want to change this response format to make it more "coin" oriented. You will understand reading the entire thread. 

We think will make things a bit clearer, but binance still uses the original structure so we're not sure. 
cc: @ttodua

Reply:
> We think will make things a bit clearer, but binance still uses the original structure so we're not sure. cc: @ttodua

It's not very unified, there is no "original structure"

Reply:
it should use the new structure if we're going to emulate `fetchTransactionFee`

Reply:
Guys, I'll mention my thought:

As always , I still think that we should unquestionably consider `network` related values everywhere. having withdraw fee i.e. '1.2345' for `usdt` doesn't make sense, because it makes sense if that fee is for `usdt->tron` or `usdt->eth` or which network. 
So, the structure should have a breaking change (inevitable, but the breaking change might be shipped with overridable `true/false` option), where the structure might be:
```
{
    'USDT': {
        'withdraw': {
            'ERC20': 1.2345,
            'TRC20': 1.2345,
        },
        'deposit': {...},
        'info': {...},
    }
}
```
however, the problem is that in real-world, only few exchanges have network-diversified responses, and they just contain one entry (without network hint). In such case,  instead of `ERC20/etc` I think we should have one key named  `UNKNOWN` (_i was thinking also `DEFAULT` but it doesnt fully reflect the reality, because it might not be `DEFAULT` network at all, and exchange might have multiple networks, so we say it's `UNKNOWN` network fee because exchange doesnt give us indication_). 
This would be the only future proof way. 

as a side thought, for compatibility, we should add base method named `this.safeTransactionFees` so it will then  generate `withdraw`, `deposit`, `info` keys into main object, which would be exactly like `safeBalance` does while it adds `free, used, total` keys in main object :
```
{
    'USDT': {...},
    ...
   'free': ...
   'used': ...
   'total': ...
}

```